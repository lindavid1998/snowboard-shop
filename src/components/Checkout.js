import React from 'react';
import CartItem from './CartItem';
import '../styles/Checkout.css';

export default function Checkout(props) {
	const { cart, removeFromCart, submitOrder, updateQuantity, totalPrice } =
		props;

	const deliveryInfo = (
		<fieldset className='delivery-info'>
			<legend>Delivery</legend>
			<div className='form-rows'>
				<div className='form-row'>
					<label htmlFor='name'>Name</label>
					<input type='text' id='name' name='name' />
				</div>

				<div className='form-row'>
					<label htmlFor='email'>Email</label>
					<input type='email' id='email' name='email' />
					<div className='error-message'>Please enter a valid email</div>
				</div>

				<div className='form-row'>
					<label htmlFor='address'>Address</label>
					<input type='text' id='address' name='address' />
				</div>

				<div className='form-row'>
					<label htmlFor='city'>City</label>
					<input type='text' id='city' name='city' />
				</div>

				<div className='form-row'>
					<label htmlFor='state'>State</label>
					<input type='text' id='state' name='state' />
				</div>

				<div className='form-row'>
					<label htmlFor='zip'>ZIP Code</label>
					<input
						type='text'
						id='zip'
						name='zip'
						pattern='^\d{5}(?:[-\s]\d{4})?$'
					/>
					<div className='error-message'>Please enter a valid ZIP</div>
				</div>
			</div>
		</fieldset>
	);

	const paymentInfo = (
		<fieldset className='payment-info'>
			<legend>Payment</legend>
			<div className='form-rows card-info'>
				<div className='form-row'>
					<label htmlFor='name-on-card'>Name on card</label>
					<input type='text' id='name' name='name-on-card' />
				</div>

				<div className='form-row'>
					<label htmlFor='card-number'>Card number</label>
					<input
						type='text'
						id='card-number'
						name='card-number'
						pattern='^[0-9]+$'
					/>
					<div className='error-message'>Please enter a valid card number</div>
				</div>

				<div className='form-row'>
					<label for='expiration-date'>Expiration date</label>
					<input
						type='month'
						id='expiration-date'
						name='expiration-date'
						min='2023-04'
						max='2030-12'
					/>
				</div>
			</div>
		</fieldset>
	);

	const items = (
		<fieldset className='checkout-cart-items'>
			<legend>Cart</legend>
			{cart.map((item) => (
				<CartItem
					item={item}
					key={item.id}
					className='cart-item checkout'
					removeFromCart={removeFromCart}
					updateQuantity={updateQuantity}
				/>
			))}
		</fieldset>
	);

	return (
		<div className='Checkout'>
			<form className='checkout-form' onSubmit={submitOrder}>
				<div className='customer-info'>
					{deliveryInfo}
					{paymentInfo}
				</div>

				<div className='order-info'>
					{cart.length !== 0 ? (
						items
					) : (
						<div className='empty-cart-message'>Your cart is empty</div>
					)}

					<div className='total'>
						<div className='total-label'>Total:</div>
						<div className='total-price'>${totalPrice}</div>
					</div>

					<button type='submit' className='btn-filled btn-place-order'>
						Place Order
					</button>
				</div>
			</form>
		</div>
	);
}
