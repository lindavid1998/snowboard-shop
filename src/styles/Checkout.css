.checkout-form {
  padding: 10px 40px 50px 40px;
  width: fit-content;
  display: grid;
  grid-template-columns: auto;
  grid-template-rows: auto;
  gap: 30px;
}

.checkout-cart-items {
  grid-area: 1 / 2 / -1 / -1;
  overflow: auto;
}

.checkout-info {
  display: flex;
  flex-direction: column;
  gap: 30px;
}

.checkout-form fieldset {
	border: 0;
  padding: 0;
}

.checkout-form legend {
	font-weight: bold;
	font-size: 1.2rem;
  padding: 0;
  margin-bottom: 10px;
}

.checkout-form input {
	background-color: #2d2d2d;
	border: 0;
	border: 1px solid #2d2d2d;
	padding: 10px;
	width: 250px;
	color: white;
  transition: 0.25s all ease-in-out;
}

input:focus {
	border: 1px solid white;
	outline: none;
}

.form-row label {
	display: block;
	margin-bottom: 3px;
  font-size: 0.8rem;
}

.form-rows {
	display: grid;
	grid-template-columns: repeat(2, 300px);
	gap: 12px;
}

.checkout-cart-items {
  display: flex;
  gap: 20px;
  flex-direction: column;
}

.order-summary {
  display: flex;
  align-items: flex-end;
  justify-content: space-between;
}

.total {
  font-size: 1.2rem;
  font-weight: bold;
}

.btn-place-order {
  position: relative;
  font-size: 1.2rem;
  padding: 0.5rem;
  right: 30%;
  background-color: rgb(0, 136, 16);
}

.btn-place-order:hover {
  background-color: rgb(0, 159, 19);
  color: white;
  cursor: pointer;
}