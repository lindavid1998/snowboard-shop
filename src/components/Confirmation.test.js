import { render, screen } from '@testing-library/react';
import Checkout from './Checkout';
import { BrowserRouter } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import { act } from 'react-dom/test-utils';
import renderer from 'react-test-renderer';

describe('Confirmation', () => {
  test.todo('renders thank you message')

	test.todo('routes to shop page');
});
