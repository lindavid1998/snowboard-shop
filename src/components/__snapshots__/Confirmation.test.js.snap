// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirmation renders thank you message 1`] = `
<div
  className="Confirmation"
>
  <h2>
    Your order is confirmed!
  </h2>
  <h3>
    Thank you
  </h3>
  <a
    href="/shop"
    onClick={[Function]}
  >
    <button
      className="btn-translate-on-hover btn-filled btn-medium"
    >
      Shop More
    </button>
  </a>
</div>
`;
